Task 2/Task 2.2/Task 2.2/Map.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+
+namespace Task_2._2
+{
+    class Map
+    {
+        Gameobj[,,] _obj;
```

Comment:
I'd use 2D array (for coordinates) containing a collection of elements - this way you could combine permeable objects with other objects without layers.